### PR TITLE
TST: Add test for `DarshanRecordCollection.to_df()`

### PR DESCRIPTION
**Pull request by nawtrey**
* Add a unit test for `DarshanRecordCollection.to_df()` to
increase confidence in performance optimizations

### CODE REVIEW

**Comment by nawtrey** [comment]
Interesting, I somehow missed a syntax error. I'll fix that and rebase/force push.

**Comment by nawtrey** [comment]
This is meant to address @tylerjereddy's [comment](https://github.com/darshan-hpc/darshan/pull/533#discussion_r734795265) in gh-533. It covers all of `to_df()` with the exception of the lines for the `LUSTRE` module, which I didn't touch because I have no example of what an "expected" outcome is.

It is a bit more leg work than I wanted to do for a test like this but I haven't found a better way of doing it. I'm certainly open to recommendations.

**Comment by tylerjereddy** [comment]
Maybe Jakob or Shane can describe the desired behavior for `LUSTRE` so that can be locked down. 

It may be a bit before I can review in more detail, but I'm not sure I saw the NumPy random seeds pinned down--you'll generally want a test to be "deterministic" instead of using different random data each time to make failures more sane to resolve later on. The old/easy way to do this was with i.e., `np.random.seed(123)`, although I believe there's a new/preferred/modern way to do this per the new NumPy random docs.

**Comment by nawtrey** [comment]
> It may be a bit before I can review in more detail, but I'm not sure I saw the NumPy random seeds pinned down--you'll generally want a test to be "deterministic" instead of using different random data each time to make failures more sane to resolve later on. The old/easy way to do this was with i.e., `np.random.seed(123)`, although I believe there's a new/preferred/modern way to do this per the new NumPy random docs.

I updated the code to use a seeded random number generator via `numpy.random.default_rng()`.

**Review by tylerjereddy — COMMENTED** [review]
Mostly seems "ok" I think. The awkwardness in the data construction may be difficult to avoid.

File: darshan-util/pydarshan/tests/test_report.py [hunk 1]
```diff
@@ -9,7 +9,14 @@
 import numpy as np
 from numpy.testing import assert_allclose
 
+rng = np.random.default_rng(seed=0)
```
[the remark this answers is not in the document]
Comment (nawtrey) [reply darshan-util/pydarshan/tests/test_report.py:12]: That's a good point. I'll move it inside of `test_to_df_synthetic()`.

[the remark this answers is not in the document]
Comment (nawtrey) [reply darshan-util/pydarshan/tests/test_report.py]: Okay, I moved this into `test_to_df_synthetic()`, so this can probably be resolved.

**Comment by tylerjereddy** [comment]
LGTM, most of the method is still covered after a local double check, merging, thanks

**Pull request merged**